client/landing/stepper/declarative-flow/copy-site.tsx

```diff
@@ -90,6 +90,18 @@ const COPY_SITE_STEPS = [
 	{ slug: 'resuming', component: ProcessingStep }, // Needs siteSlug param
 ];
 
+const getValidCopyPlanFromSearchParams = ( urlQueryParams: URLSearchParams ) => {
+	const plan = urlQueryParams.get( 'plan' );
+	switch ( plan ) {
+		case 'ecommerce-bundle':
```

Comment:
This only works for the yearly bundle. If add a 3y, 2y or a monthly subscription I get a business plan.

It's ok-ish for business because it defaults to business yearly. However, it should still take into account the plan recurrence (if the user opted for a monthly plan, it should still point to a monthly plan).

Reply:
if we want to match the plan length on the existing site then there's a bunch of prior art hanging around [e.g.](https://github.com/Automattic/wp-calypso/blob/0855b873737608ee0a03a82f9d48d27cfc411842/client/my-sites/themes/theme-options.js#L69)